config/migrate.go

```diff
@@ -198,3 +198,25 @@ func getVersionedDefaultLocalConfig(version uint32) (local Local) {
 	}
 	return
 }
+
+// GetNonDefaultConfigValues takes a provided cfg and list of field names, and returns a map of all values in cfg
+// that are not set to the default for the latest version.
+func GetNonDefaultConfigValues(cfg Local, fieldNames []string) map[string]interface{} {
```

Comment:
Could you change this into an array of values so that it doesn't add to the ES mapping?

Reply:
OK I think you mean like an array of `[]struct{Name string, Value interface{}}`.. sure

Reply:
updated

Reply:
I was excited to look for anyone overriding a specific field but I think it is possible whether or not we use an array or nested doc here